python/xorbits/_mars/dataframe/groupby/aggregation.py

```diff
@@ -1017,12 +1018,22 @@ def _do_predefined_agg(input_obj, agg_func, single_func=False, **kwds):
 
         if ndim == 2:
             if single_func:
-                result = input_obj.agg(agg_func)
+                # DataFrameGroupby.agg('size') returns empty df in cudf, which is not correct
+                # The index order of DataFrameGroupby.size() is wrong in cudf
+                result = (
+                    input_obj.size().sort_index()
+                    if gpu and agg_func == "size"
+                    else input_obj.agg(agg_func)
+                )
                 if result.ndim == 1:
                     # when agg_func == size, agg only returns one single series.
                     result = result.to_frame(agg_func)
             else:
-                result = input_obj.agg([agg_func])
+                result = (
+                    input_obj.agg([agg_func]).sort_index()
```

Comment:
`sort_index` would affect performance, for tests, we can sort manually after executing.